superset/assets/src/SqlLab/components/ScheduleQueryButton.jsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Form from 'react-jsonschema-form';
+import { t } from '@superset-ui/translation';
+
+import Button from '../../components/Button';
+import ModalTrigger from '../../components/ModalTrigger';
+
+const propTypes = {
+  defaultLabel: PropTypes.string,
+  sql: PropTypes.string,
+  schema: PropTypes.string,
+  dbId: PropTypes.number,
+  animation: PropTypes.bool,
+  onSchedule: PropTypes.func,
+};
+const defaultProps = {
+  defaultLabel: t('Undefined'),
```

Comment:
Should we provide default values for the other props?

Reply:
Ah, good point. I'll make the other ones as required (sql, schema, dbid).